notebooks/ch17/classification_sensitivity_specificity.md

```diff
@@ -0,0 +1,486 @@
+
+* `ham_only`: $\frac{0}{0 + 42} = 0$
+* `spam_only`: $\frac{42}{42 + 0} = 1$
+* `words_list_model`: $\frac{18}{18 + 24} \approx .429$
+
+Since `ham_only` has no true positives, it has the worst possible sensitivity value of 0. On the other hand, `spam_only` has an abysmally low accuracy but it has the best possible sensitivity value of 1. The low sensitivity of `words_list_model` indicates that it frequently fails to mark spam emails as such. Nonetheless, it outperforms `ham_only`, which will prove significant when we examine ROC curves later.
+
+### Specificity
+
+**Specificity** (also called **true negative rate**) measures the proportion of data belonging to the negative class that the classifier correctly labels.
+
+$$ \text{Precision} = \frac{TN}{TN + FP} $$ 
+
+We can use values from the confusion matrices to compare the specificities of our models:
+
+* `ham_only`: $\frac{758}{758 + 0} = 1$
+* `spam_only`: $\frac{0}{0 + 758} = 0$
+* `words_list_model`: $\frac{752}{752 + 6} \approx .992$
+
+As with sensitivity, the worst and best specificities are 0 and 1 respectively. Notice that `ham_only` has the best specificity and worst sensitivity, while `spam_only`has the worst specificity and best sensitivity. Since these models only predict one label, they will misclassify all instances of the other label, which is reflected in the extreme sensitivity and specificity values. This contrast is not nearly as pronounced for `words_list_model`. 
+
+Both sensitivity and specificity are needed to thoroughly evaluate a model, but it is not yet clear how we can use them to conclusively determine which model is better. Before we delve into this, we will take a slight detour to discussion classification threshold.
+
+### Classification Threshold
+
+The **classification threshold** is a value that determines what class a data point is assigned to; points that fall on opposite sides of the threshold are labeled with different classes. Recall that logistic regression outputs a probability that the data point belongs to the positive class. If the probability output on $x_1$ is .90 and the output on $x_2$ is .1, then we can be fairly confident that $x_1$ is spam and $x_2$ is ham. However, if the probability associated with $x_3$ is around .55, there is greater uncertainty about the correct class. If the classification threshold is .50, $x_3$ is labeled as spam, but if the threshold is .60, $x_3$ is labeled as ham.
+
+Although the goal is to maximize both sensitivity and specificity, they are inversely related so there is a tradeoff. Increasing sensitivity leads to a decrease in specificity and vice versa. For instance, reducing false negatives by lowering the classification threshold increases false positives on data that initially fell near the classification threshold but was correctly labeled with the negative class. 
```

Comment:
Instead of talking about sensitivity and specificity straight away, can you provide a short example of adjusting the classification threshold between ~3 values and showing how the confusion matrix changes? This will make the trade-off between sensitivity and specificity much clearer (since we "show, not tell") and provide better motivation for the ROC curve.